File: src/Duck.cpp
#include "Duck.hpp"
#include <iostream>

using namespace std;

Duck::Duck() :FarmAnimal(false, true, false, 12,0, 0) {
}

Duck::Duck(int x, int y):FarmAnimal(false, true, false, 12,x, y) {
}

void Duck::Bersuara() //Menghasilkan suara Duck
{
    cout<<"Kwek Kwek"<<endl; 
}

char Duck::render() //Menghasilkan representasi dari Duck
{
    return 'B';
}

FarmProduct* Duck::interact() //Berinteraksi dengan Duck menghasilkan DuckEgg
{
    return new DuckEgg();
} 

File: src Java/Duck.java
public class Duck extends FarmAnimal {
    /**
     * Default constructor
     */
    public Duck(){
        super(false, true, false, 12, 0, 0);
    }

    /**
     * User-defined constructor
     * @param x posisi X hewan yang diinginkan
     * @param y posisi Y hewan yang diinginkan
     */
    public Duck(int x, int y){
        super(false, true, false, 12, x, y);
    }

    /**
     * Mencetak representasi suara bebek
     */
    public void bersuara(){
        System.out.println("Kwek kwek");
    }

    /**
     * Mengembalikan karakter representasi bebek
     * @return 'B'
     */
    public char render(){
        return 'B';
    }

    /**
     * Mengembalikan FarmProduct ketika pemain berinteraksi dengan bebek
     * @return objek DuckEgg
     */
    public FarmProduct interact(){
        return new DuckEgg();
    }
}
